Gemfile

```diff
@@ -32,6 +31,11 @@ group :development, :test do
   gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
   gem "capybara", ">= 2.15"
   gem "selenium-webdriver"
+  gem "rspec-rails"
+  gem "factory_bot_rails"
+  gem "faker", :git => "https://github.com/stympy/faker.git", :branch => "master"
+  gem "shoulda-matchers", "~> 3.1"
```

Comment:
gem này để làm gì thế em?

Reply:
@huongnguyenmta 
Shoulda matchers để giúp việc test các chức năng dễ dàng hơn thay vì viết bằng tay sẽ dài, phức tạp và dễ bị lỗi. Những matchers mà shoulda matchers hỗ trợ có (ActiveModel, ActiveRecord, ActiveController)
Nó còn có các lớp tương thích vs RSpec và Minitest.

Reply:
Lại trả lời kiểu TRẢ BÀI rồi
Khi tìm hiểu về gem thì em tìm hiểu rồi trả lời 3 câu hỏi sau nhé:
- Tác dụng
- Cách dùng + các method phổ biến thường dùng
- Cơ chế hoạt động

Reply:
@huongnguyenmta 
- Tác dụng:
+) Shoulda Matchers cung cấp cách viết 1 dòng lệnh Rspec dùng để test các chức năng của Rails.
+) Sử dụng shoulda matchers giúp viết test 1 cách ngắn gọn, rõ ràng và ít gặp lỗi hơn viết tay.
- Cách dùng: should method hoặc is_expected.to method
- Các method thường dùng:
Active Model:
+) allow_value
+) validate_presence_of
+) validate_length_of
+) have_secure_password
ActiveRecord:
+) accept_nested_attributes_for
+) belong_to 
+) have_db_column
+) have_db_index
+) have_many
ActiveController: 
+) redirect_to
+) render_template
+) use_before/after_action
- Hoạt động:
+) should method hoạt động kể cả khi sử dụng cú pháp expect.
+) is_expected hoạt động độc lập với expect trong it.

Reply:
có chút sơ hở về Cơ chế hoạt động nhé
Cơ mà em về tìm hiểu thêm nhé